VirtoCommerce.Storefront/Domain/Cart/CartBuilder.cs

```diff
@@ -578,17 +624,23 @@ protected virtual Task ChangeItemQuantityAsync(LineItem lineItem, int quantity)
         {
             if (lineItem != null && !lineItem.IsReadOnly)
             {
+
                 if (lineItem.Product != null)
                 {
-                    var salePrice = lineItem.Product.Price.GetTierPrice(quantity).Price;
-                    if (salePrice != 0)
-                    {
-                        lineItem.SalePrice = salePrice;
-                    }
-                    //List price should be always greater ot equals sale price because it may cause incorrect totals calculation
-                    if (lineItem.ListPrice < lineItem.SalePrice)
+                    var product = lineItem.Product;
+
+                    if (product.Price.ListPrice == lineItem.ListPrice)
```

Comment:
This strange condition is required your clarification.

Reply:
This condition can be removed when 'custom price'  attribute will be added to the line item. this attribute means that price validation shouldn't need.